src/pad_constant_api.cpp

```diff
@@ -0,0 +1,71 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/errors.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/logger.hpp>
+#include <miopen/tensor_ops.hpp>
+#include <miopen/pad_constant.hpp>
+
+extern "C" miopenStatus_t miopenPadConstantFwd(miopenHandle_t handle,
+                                               miopenTensorDescriptor_t xDesc,
+                                               miopenTensorDescriptor_t yDesc,
+                                               const void* x,
+                                               void* y,
+                                               const size_t* padding,
+                                               float value)
+{
+    MIOPEN_LOG_FUNCTION(handle, xDesc, x, padding, value, y);
+
+    return miopen::try_([&] {
+        miopen::PadConstantForward(miopen::deref(handle),
+                                   miopen::deref(xDesc),
+                                   miopen::deref(yDesc),
+                                   DataCast(x),
+                                   DataCast(y),
+                                   padding,
+                                   value);
+    });
+}
+
+extern "C" miopenStatus_t miopenPadConstantBwd(miopenHandle_t handle,
+                                               miopenTensorDescriptor_t xDesc,
+                                               miopenTensorDescriptor_t yDesc,
+                                               void* x,
+                                               const void* y,
+                                               const size_t* padding)
+{
+    MIOPEN_LOG_FUNCTION(handle, xDesc, x, y, padding);
+
+    return miopen::try_([&] {
+        miopen::PadConstantBackward(miopen::deref(handle),
+                                    miopen::deref(xDesc),
+                                    miopen::deref(yDesc),
+                                    DataCast(x),
+                                    DataCast(y),
+                                    padding);
```

Comment:
use miopen::deref for pointer.

Reply:
It's about padding. MIOpen always uses miopen::deref for pointers received from the user.

Reply:
You can solve this by using **. However, when I checked other functions of MIOpen, I found that int and size_t do not use dref, so you do not need to worry about them.